src/cache.js

```diff
@@ -0,0 +1,58 @@
+import { endGroup, getInput, startGroup, warning } from '@actions/core';
+import { restoreCache, saveCache } from '@actions/cache';
+import { hashFiles } from '@actions/glob';
+import { context } from '@actions/github';
+import { downloadLatestArtifact, getCurrentWorkflowId } from './workflow';
+import { Options } from './options';
+
+/**
+ * @param {WorkingDirectory} workingDirectory
+ * @returns {Promise<string[]>}
+ */
+export async function computeCacheKey(workingDirectory) {
+    const hash = await hashFiles(workingDirectory.options);
+    const key0 = `packsquash-0-${getInput(Options.ActionCacheRevision)}-${getInput(Options.PackSquashVersion)}-${hash}`;
```

Comment:
I think it's still important to sanitize the value of the cache revision parameter, as it is a free-form string that may contain forbidden characters for a cache key. The shell script sanitized this value with the following code, but you can implement any other transformation that you'd like (a hash is not advised due to the possibility of collisions):

https://github.com/ComunidadAylas/PackSquash-action/blob/cd8aaf4b20f3634d35aba76a5e7b46acf0e92cc2/entrypoint.sh#L194-L197

Reply:
Fix by e423c2504f5531648b7ac3f52c7dc173d379c1d4

According to the source code, the only prohibited character is a comma, so I chose base64.
https://github.com/actions/toolkit/blob/7654d97eb6c4a3d564f036a2d4a783ae9105ec07/packages/cache/src/cache.ts#L32-L44